gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -476,20 +497,18 @@ end
 ---
 -- Gives an @{ITEM} to a @{Player} and returns whether it was successful
 -- @param string cls
--- @return boolean success?
+-- @return @{ITEM} or false
 -- @realm server
 function plymeta:GiveEquipmentItem(cls)
-	if not cls then return end
+	if not cls then return false end
 
 	local item = items.GetStored(cls)
 
 	if not item or item.limited and self:HasEquipmentItem(cls) then
 		return false
 	end
 
-	self:AddEquipmentItem(cls)
-
-	return true
+	return self:AddEquipmentItem(cls)
```

Comment:
we have to talk about this. Why return true/false and not the item? Maybe some addon relies on the returned item? A simple IsValid check solves the problem and I see no need in switching to booleans here

Reply:
I think you got it backwards. Before the change it was always a boolean value, now it is the item on success. The function never returns true and instead returns the item.